expression/integration_test.go

```diff
@@ -5436,6 +5436,13 @@ func (s *testIntegrationSuite) TestCollation(c *C) {
 	tk.MustQuery("select collation('a')").Check(testkit.Rows("utf8mb4_bin"))
 	tk.MustExec("set names utf8mb4 collate utf8mb4_general_ci")
 	tk.MustQuery("select collation('a')").Check(testkit.Rows("utf8mb4_general_ci"))
+
+	tk.MustExec("set names utf8mb4 collate utf8mb4_general_ci")
+	tk.MustExec("set @test_collate_var = 'a'")
+	tk.MustQuery("select collation(@test_collate_var)").Check(testkit.Rows("utf8mb4_general_ci"))
+	tk.MustExec("set names utf8mb4 collate utf8mb4_general_ci")
+	tk.MustExec("set @test_collate_var = 1")
+	tk.MustQuery("select collation(@test_collate_var)").Check(testkit.Rows("utf8mb4_general_ci"))
```

Comment:
Please add the following test:
set @test_collate_var = 'a' collate utf8mb4_bin;
set @test_collate_var = concat("a", "b" collate utf8mb4_bin);

Reply:
for `set @test_collate_var = 'a' collate utf8mb4_bin`,
I can't get `utf8mb4_bin` from `VarAssignment.Value` or `VarAssignment.ExtendValue` In `SetExecutor.Next()`.
May be it is because of this syntax is not yet supported in `parser`.